algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_model_ucapi_impl.h

```diff
@@ -0,0 +1,179 @@
+
+    /**
+     * Returns training data
+     * \return Training data
+     */
+    data_management::NumericTablePtr getData() { return _data; }
+
+    template<typename Archive, bool onDeserialize>
+    services::Status serialImpl(Archive * arch)
+    {
+        arch->setSharedPtrObj(_data);
+        arch->setSharedPtrObj(_labels);
+
+        return services::Status();
+    }
+
+    /**
+     * Sets a training data
+     * \param[in]  value  Training data
+     * \param[in]  copy   Flag indicating necessary of data deep copying to avoid direct usage and modification of input data.
+     */
+    template <typename algorithmFPType>
+    DAAL_EXPORT DAAL_FORCEINLINE services::Status setData(const data_management::NumericTablePtr & value, bool copy)
+    {
+        if (!copy)
+        {
+            _data = value;
+        }
+        else
+        {
+            services::Status status;
+            _data = SyclHomogenNumericTable<algorithmFPType>::create(value->getNumberOfColumns(), value->getNumberOfRows(),
+                                                                     NumericTable::doAllocate, &status);
+            DAAL_CHECK_STATUS_VAR(status);
+            status |= _data->allocateDataMemory();
+            data_management::BlockDescriptor<algorithmFPType> destBD, srcBD;
+            status |= _data->getBlockOfRows(0, _data->getNumberOfRows(), data_management::writeOnly, destBD);
+            status |= value->getBlockOfRows(0, value->getNumberOfRows(), data_management::readOnly, srcBD);
+            DAAL_CHECK_STATUS_VAR(status);
+            auto source = srcBD.getBuffer();
+            auto destination = destBD.getBuffer();
+            auto& context = Environment::getInstance()->getDefaultExecutionContext();
+            context.copy(destination, 0, source, 0, source.size(), &status);
+            DAAL_CHECK_STATUS_VAR(status);
+            status |= _data->releaseBlockOfRows(destBD);
+            status |= value->releaseBlockOfRows(srcBD);
+            DAAL_CHECK_STATUS_VAR(status);
+        }
+        return services::Status();
+    }
+
+    /**
+     * Returns training labels
+     * \return Training labels
+     */
+    data_management::NumericTableConstPtr getLabels() const { return _labels; }
+
+    /**
+     * Returns training labels
+     * \return Training labels
+     */
+    data_management::NumericTablePtr getLabels() { return _labels; }
+
+    /**
+     * Sets a training data
+     * \param[in]  value  Training labels
+     * \param[in]  copy   Flag indicating necessary of data deep copying to avoid direct usage and modification of input labels.
+     */
+    template <typename algorithmFPType>
+    DAAL_EXPORT DAAL_FORCEINLINE services::Status setLabels(const data_management::NumericTablePtr & value, bool copy)
+    {
+        if (!copy)
+        {
+            _labels = value;
+        }
+        else
+        {
+            services::Status status;
```

Comment:
The code in `setData` and `setLabels` methods looks the same. Suppose we can extract another method here